TransmissionVPN: Initial Role Addition

tarek369 opened the pull request.

tarek369 commented:
When it will be merged?

Migz93 commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,83 @@
+  include_role:
+    name: cloudflare-dns
+  vars:
+    subdomain: transmission
```
Change the subdomain to "transmissionvpn"

tarek369 replied:
done

Migz93 commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,83 @@
+
+- name: Stop and remove any existing container
+  docker_container:
+    name: transmission
```
Change the stopped container to "transmissionvpn"

tarek369 replied:
done

Migz93 commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,83 @@
+    env:
+      PUID: "{{ uid }}"
+      PGID: "{{ gid }}"
+      VIRTUAL_HOST: "transmission.{{ user.domain }}"
```
Change virtual host to "transmissionvpn"

tarek369 replied:
done

Migz93 commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,83 @@
+      PGID: "{{ gid }}"
+      VIRTUAL_HOST: "transmission.{{ user.domain }}"
+      VIRTUAL_PORT: 9091
+      LETSENCRYPT_HOST: "transmission.{{ user.domain }}"
```
Change letsencrypt host to "transmissionvpn"

tarek369 replied:
done

Migz93 commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,83 @@
+    networks:
+      - name: cloudbox
+        aliases:
+          - transmission
```
Change alias to "transmissionvpn"

tarek369 replied:
done

Migz93 commented:
A couple of changes to make to specify "transmissionvpn" in a few places instead of "transmission" and then i can merge.

tarek369 commented:
When this one will be merged?

Migz93 commented:
Are you able to rebase this with the latest dev? Unfortunately this PR is a little out of date, through no fault of your own.

Also is anyone else able to test tareks fork and confirm it works as expected? Unfortunately i don't have a VPN to try with.

tarek369 commented:
I will tomorrow, and for the VPN I tested from nordvpn and custom ! So custom is basically normal vps with openvpn installed , I can give you access for it if you want to test it.

tarek369 commented:
Done ! send me your email to send you the vpn

Migz93 commented:
It's okay, i've tried it with a "CUSTOM" provider, ovpn file provided by my usenet.
It worked fine, will need a wiki page to explain how to setup though.

Just wondering, what is the location on line 38 in "main.yml" for? A random extra vpn folder in the download location, is this required?

tarek369 commented:
No it's seems mistake ! 
For the wiki page , it's actually much easier when I tried with Nord I just added my user and password in settings.yml and it downloaded the file automatically .. and also I added the link for the original repo

Migz93 commented:
Just remove that line 38 then and from my view everything is good to go.

tarek369 commented:
Done

Migz93 commented:
@desimaniac could you please give one last eye over this PR? Everything looks fine to me, tested it and confirmed it's working at least with the custom option.

desimaniac commented on roles/transmissionvpn/tasks/main.yml:
```diff
@@ -0,0 +1,82 @@
+- name: Set default_volumes variable
+  set_fact:
+    default_volumes:
+      - "/etc/localtime:/etc/localtime:ro"
```
Take this out and use TZ env variable instead. Lot of issues with mounting this (ie `error: mounting folder as a file`, etc), which is why I made `{{ tz }}` a global variable to be used for docker env TZ.

desimaniac replied:
@Migz93 just one change i see that needs to be made.

tarek369 replied:
Im not sure exactly what that means! @Migz93 it will be nice if you can make it or you can explain to me and I will do it :)

Migz93 replied:
Not sure how to modify your PR directly. But i've made the changes here
https://pastebin.com/HW4yE5sc

Basically i removed this line (Line 44), from default_volumes
`- "/etc/localtime:/etc/localtime:ro"`
And instead i added this line (Line 70), to set TZ as a env option, pulling the variable from the users adv_settings file
`TZ: "{{ tz }}"`

tarek369 replied:
done
